src/emsdevicevalue.cpp

```diff
@@ -331,7 +331,7 @@ bool DeviceValue::get_custom_min(int16_t & val) {
         if (fahrenheit) {
             v = (v - (32 * (fahrenheit - 1))) / 1.8; // reset to °C
         }
-        if (max > 0 && v > max) {
+        if (max > 0 && v > (int16_t)max) {
```

Comment:
Could be wrong to shorten max, if you want to compare identical types do 
`if (max > 0 && v > 0 && (uint32_t)v >max) {`

Maybe it's much easier to set max as int32_t and use 31 bit range and handle the inputs for min/max also in int32_t, only shorten min to int16_t for storage.

Reply:
thanks, this was just so the code would compile on my MacBook - the compiler would throw an error. I'll go for option 1 or look at a compiler option to ignore these checks.